pkgs/by-name/av/avalanche-cli/package.nix

```
@@ -0,0 +1,58 @@
+{
+  stdenv,
+  fetchFromGitHub,
+  lib,
+  buildGoModule,
+  blst,
+  libusb1,
+  installShellFiles,
+  nix-update-script,
+}:
+buildGoModule (finalAttrs: rec {
+  pname = "avalanche-cli";
+  version = "1.9.0";
+
+  src = fetchFromGitHub {
+    owner = "ava-labs";
+    repo = "avalanche-cli";
+    tag = "v${version}";
+    sha256 = "sha256-LLGNeKlZVGztrgap8Ih6BMLlMCp7acAnxb7zcExixNA=";
+  };
+
+  proxyVendor = true;
+  vendorHash = "sha256-F2prsymg2ean7Er/tTYVUrdyOdtMhxk5/pyOJzONrr8=";
+
+  doCheck = false;
+
+  ldflags = [
+    "-X=github.com/ava-labs/avalanche-cli/cmd.Version=${version}"
+  ];
+
+  buildInputs = [
+    blst
+    libusb1
+  ];
+
+  nativeBuildInputs = [ installShellFiles ];
+  postInstall = ''
+    mv $out/bin/avalanche-cli $out/bin/avalanche
+
+    ${lib.optionalString (stdenv.buildPlatform.canExecute stdenv.hostPlatform) ''
+      installShellCompletion --cmd avalanche \
+        --bash <($out/bin/avalanche completion bash) \
+        --fish <($out/bin/avalanche completion fish) \
+        --zsh <($out/bin/avalanche completion zsh)
+    ''}
```

Non-blocking suggestion: there's a more cross-friendly way to generate completions
```suggestion
    ${lib.optionalString (stdenv.hostPlatform.emulatorAvailable buildPackages) (
      let
        emulator = stdenv.hostPlatform.emulator buildPackages;
      in
      ''
        installShellCompletion --cmd avalanche \
          --bash <(${emulator} $out/bin/avalanche completion bash) \
          --fish <(${emulator} $out/bin/avalanche completion fish) \
          --zsh <(${emulator} $out/bin/avalanche completion zsh)
      ''
    )}
```
Source: https://nixos.org/manual/nixpkgs/stable/#installshellfiles-installshellcompletion-exampleusage-guarded